src/main/scala/com/abdulradi/opentracing/xray/v1/Format.scala

```diff
@@ -56,11 +62,39 @@ object Format {
       "cause" -> obj.cause.asJson
     )
   )
-  implicit val causeEncoder: ObjectEncoder[Cause] = dummy // TODO
-  implicit val causeObjectEncoder: ObjectEncoder[CauseObject] = dummy // TODO
-  implicit val exceptionDetailsEncoder: ObjectEncoder[ExceptionDetails] = dummy // TODO
-  implicit val stackFrameEncoder: ObjectEncoder[StackFrame] = dummy // TODO
+  implicit val causeEncoder: Encoder[Cause] = new Encoder[Cause]() {
+    def apply(value: Cause) = value match {
+      case ExceptionId(value) => value.asJson
```

Comment:
There is a bug in the model. `CauseObject` should extend from `Cause`, and of course should be handled here.
Can you please fix it?

Reply:
sure